lib/paperclip/attachment.rb

```diff
@@ -613,5 +614,21 @@ def able_to_store_created_at?
     def has_enabled_but_unset_created_at?
       able_to_store_created_at? && !instance_read(:created_at)
     end
+
+    # Checks whether the option to show attributes to be destroyed is enabled and whether a destroy callback has been
+    # invoked when deleting the entire record.
+    def show_attrs_and_destroy_callback_triggered?
+      @options[:return_file_attributes_on_destroy] && instance.instance_variable_get(:@_destroy_callback_already_called)
```

Comment:
why do we have the second condition?

Reply:
First, we check whether the option is enabled (for backwards compatibility) and then whether the destroy callback got triggered.